File: rust/src/models/plan_event.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{event_id::EventId, event_type::EventType, plan_event_data::PlanEventData};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct PlanEvent {
    #[serde(flatten)]
    pub flatten_planeventdata: PlanEventData,

    pub id: EventId,

    pub timestamp: String,

    pub r#type: EventType,
}

impl PlanEvent {
    pub fn new(id: EventId, timestamp: String, r#type: EventType) -> Self {
        Self {
            flatten_planeventdata: Default::default(),
            id,
            timestamp,
            r#type,
        }
    }
}

File: typescript/src/models/planEvent.ts
// this file is @generated
import { parseDateTime } from "../datetime";
import { type EventId, EventIdSerializer } from "./eventId";
import { type EventType, EventTypeSerializer } from "./eventType";
import { type PlanEventData, PlanEventDataSerializer } from "./planEventData";

export interface PlanEvent {
  flattenPlaneventdata: PlanEventData;

  id: EventId;

  timestamp: Date;

  type: EventType;
}

export const PlanEventSerializer = {
  _fromJsonObject(object: any): PlanEvent {
    return {
      flattenPlaneventdata: PlanEventDataSerializer._fromJsonObject(
        object["__flatten_planeventdata"]
      ),
      id: EventIdSerializer._fromJsonObject(object["id"]),
      timestamp: parseDateTime(object["timestamp"]),
      type: EventTypeSerializer._fromJsonObject(object["type"]),
    };
  },

  _toJsonObject(self: PlanEvent): any {
    return {
      __flatten_planeventdata: PlanEventDataSerializer._toJsonObject(
        self.flattenPlaneventdata
      ),
      id: EventIdSerializer._toJsonObject(self.id),
      timestamp: self.timestamp,
      type: EventTypeSerializer._toJsonObject(self.type),
    };
  },
};
